src/nwb2bids/_converters/_dataset_converter.py

```diff
@@ -154,19 +227,32 @@ def convert_to_bids_dataset(
             - "symlink": Create symbolic links to the files in the BIDS directory.
             - "auto": Decides between all the above based on the system, with preference for linking when possible.
         """
-        bids_directory = self._handle_bids_directory(bids_directory=bids_directory)
+        try:
+            bids_directory = self._handle_bids_directory(bids_directory=bids_directory)
 
-        if self.dataset_description is not None:
-            self.write_dataset_description(bids_directory=bids_directory)
+            if self.dataset_description is not None:
+                self.write_dataset_description(bids_directory=bids_directory)
 
-        self.write_participants_metadata(bids_directory=bids_directory)
-        self.write_sessions_metadata(bids_directory=bids_directory)
+            self.write_participants_metadata(bids_directory=bids_directory)
+            self.write_sessions_metadata(bids_directory=bids_directory)
 
-        generator = (
-            session_converter.convert_to_bids_session(bids_directory=bids_directory, file_mode=file_mode)
-            for session_converter in self.session_converters
-        )
-        collections.deque(generator, maxlen=0)
+            generator = (
+                session_converter.convert_to_bids_session(bids_directory=bids_directory, file_mode=file_mode)
+                for session_converter in self.session_converters
+            )
+            collections.deque(generator, maxlen=0)
+        except:  # noqa
```

Comment:
```suggestion
        except Exception:  # noqa
```

The bare `except` would catch `SystemExit` and `KeyboardInterrupt` as well.